Marlin/Conditionals_post.h

```diff
@@ -780,11 +780,11 @@
   #endif
 
   #if IS_KINEMATIC
-    // Check for this in the code instead
-    #define MIN_PROBE_X X_MIN_POS
-    #define MAX_PROBE_X X_MAX_POS
-    #define MIN_PROBE_Y Y_MIN_POS
-    #define MAX_PROBE_Y Y_MAX_POS
+    // Check for this in the code instead (preprocessor can't do floating point arithmetic)
+    #define MIN_PROBE_X (-10000)
+    #define MAX_PROBE_X ( 10000)
+    #define MIN_PROBE_Y (-10000)
+    #define MAX_PROBE_Y ( 10000)
```

Comment:
The problem here is that MIN_PROBE_X and friends are used with preprocessor math, meaning they cannot be float.  But on delta, DELTA_PRINTABLE_RADIUS is a float (at least in all the sample delta config files), and MIN_PROBE_X etc are defined relative to DELTA_PRINTABLE_RADIUS.

So two options -- either require DELTA_PRINTABLE_RADIUS to be integer which might break existing user config.h or ignore all these preprocessor checks for delta and require the user to specify manual MIN_PROBE_X etc that are integers without being able to preprocessor-verify that they are within float DELTA_PRINTABLE_RADIUS, which is kind of silly anyway because the corners are still unreachable, but that is handled now within the code.

For deltas, the true probeable area is actually a circle (printable_radius) minus a crescent (offset of probe from nozzle).  This is reflected in the new position_is_reachable_by_probe for deltas.  So delta won't probe off the bed (at least for UBL) even if the MIN_PROBE_X etc are outside the radius.  

Personally I think MIN_PROBE_X etc should simply be undefined for DELTA completely and leave that definition/verification mechanism for cartesian-only.  Delta probe "grid" for UBL can be defined as the square that encompasses the printable radius.  For other bed leveling systems on delta (I think BILINEAR is the only one) I'm not certain but I think the probabeable rectangle must fit completely within the printable and probeable region of the bed.  I could change that, but it would fall outside UBL_DELTA semantics.

I'm open to suggestions.